x-pack/plugins/maps/public/shared/layers/sources/es_source.js

```diff
@@ -45,6 +46,13 @@ export class AbstractESSource extends AbstractVectorSource {
     this._inspectorAdapters.requests.resetRequest(this._descriptor.id);
   }
 
+  cloneDescriptor() {
+    const clonedDescriptor = copyPersistentState(this._descriptor);
+    // id used as uuid to track requests in inspector
+    clonedDescriptor.id = uuid();
```

Comment:
Wondering if we need to persist this in the saved-object at all, and for es-sources we could just precede this with `.__id`? Would this mess up the inspector after clicking the `save` button?

Reply:
`id` needs to stay the same for an Elasticsearch source so the requests for the source can be updated properly in the inspector. The source descriptor is turned into Object instances many times as the store state changes and the `id` needs to stay the same across those instances